CHANGELOG.md

```diff
@@ -47,6 +47,8 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
   Users can configure it with `RPMBuilder::build_time()`.
 - Improved support for packages >4gb
 - Always write tags in sorted order
+- Changed signature for `RPMPackage::parse`. Now it apply `io::Read` instead `io::BufRead`
```

Comment:
I'm a little conflicted on this, looser constraints are usually nice but buffered IO is something you generally want to encourage to avoid your code being much slower than expected.  And we're doing a bunch of small reads to parse the headers.  With BufReader, the whole package is 99% likely to fit into the buffer (8mb default) so it's all coalesced into one actual read syscall.

Reply:
@dralley I totally agree! But, This does not mean that library cannot work with `io::Read`. If we can give options for work with simple `io::Read` than why not) U
ser can make a decision use `io::Read` or `io::BufRead`. For example, I use my own impl of working with file (If file is smaller then 5mb then I load it into memory) and then call `RPMPackage::parse`. I don't need to use `io::BufRead` but I have to wrap this into `io::BufRead` for working with RPMPackage.

Reply:
It's ok. If programmer care about performance  than he (or she) cans use `io::BufReader` and everything continue working)  It's not a breaking change)

Reply:
https://github.com/rpm-rs/rpm/blob/f6e62aa3c2f27130e44beff8f0c91a3bdcef7bc5/src/rpm/package.rs#L55 here we are working with `io::BufReader`. This PR is about convenience, not about performance)

Reply:
> For example, I use my own impl of working with file (If file is smaller then 5mb then I load it into memory) 

That doesn't seem much different from just using a `BufReader` with a size of 5mb?  Are you trying to keep the memory really low or something?

I'm fine with all of that but my misgiving is that if we later find a use case where it would have been nice to use `BufRead`, which we might depending on how we implement the whole not-loading-the-payload-immediately thing, then we'd need to change the signature back.

Reply:
I cannot use `BufRead`. My code looks like:

```rust
enum Storage {
    Memory(Vec<u8>),
    Mmap(Mmap),
}

pub struct CachedFile(Cursor<Storage>);

impl AsRef<[u8]> for CachedFile {...}
impl Read for CachedFile {...}
impl Seek for CachedFile {...}
```

I need to support both interfaces: buffer and reader.

Reply:
Just to correct something above - the internal buffer for BufReader is 8KB, not 8MB. 8KB is less likely to hold an entire RPM package.